### ISSUE TITLE
linter: `vue/no-required-prop-with-default` false positive

### ISSUE THREAD

**Issue by Sysix** [oxc-project/oxc#14056]
### What version of Oxlint are you using?

1.17.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": ["vue"],
  "rules": {
    "vue/no-required-prop-with-default": "error"
  }
}
```

### What happened?

This `laps` property should not be flagged as an error.
```vue
<script setup lang="ts">
    const { laps } = defineProps<{
        laps: unknown[];
  }>();
</script>
```

Reported in discord: https://discord.com/channels/1079625926024900739/1080712072012238858/1420085140385829015

<img width="1095" height="157" alt="vscode screenshot" src="https://github.com/user-attachments/assets/2ca03c8b-80ff-46ca-a3e1-b59982a7ca7c" />

**Comment by Perdolique** [comment]
This one is also false positive:

```ts
interface Props {
  readonly phoneNumber: string;
}

const { phoneNumber } = defineProps<Props>();
```

```shell
  × eslint-plugin-vue(no-required-prop-with-default): Prop "phoneNumber" should be optional.
    ╭─[component.vue:19:5]
 18 │   interface Props {
 19 │     readonly phoneNumber: string;
    ·     ─────────────────────────────
 20 │   }
    ╰────
  help: Remove the `required: true` option, or drop the `required` key entirely to make this prop optional.
```
